### PR TITLE
enhance: add an unify vector index config checker

### PR DESCRIPTION
**Pull request by foxspy**
issue: #34298

### CODE REVIEW

**Comment by foxspy** [comment]
rerun go-sdk

**Comment by foxspy** [comment]
rerun go-sdk

**Comment by foxspy** [comment]
racket

**Comment by liliu-z** [comment]
@foxspy 
[pytest : test] [2024-10-23 16:48:58 - ERROR - pymilvus.decorators]: RPC error: [create_index], <MilvusException: (code=1100, message=metric type not found or not supported, supported: [IP]: invalid parameter[expected=valid index params][actual=invalid index params])>, <Time:{'RPC start': '2024-10-23 16:48:58.544908', 'RPC error': '2024-10-23 16:48:58.547414'}> (decorators.py:140)

**Review by liliu-z — COMMENTED** [review]
/lgtm

File: internal/util/indexparamcheck/index_type.go [hunk 1]
```diff
@@ -23,31 +24,7 @@ type IndexType = string
...
-	IndexSparseWand      IndexType = "SPARSE_WAND"
-	// For temporary use, will be removed in the future.
-	// 1. All Index related param check will be moved to Knowhere recently.
-	// 2. FAISS_HNSW_xxx will be rename to HNSW_xxx after QA test. We keep the original name for comparison purpose.
-	// TODO: @liliu-z @foxspy
-	IndexFaissHNSW    IndexType = "FAISS_HNSW_FLAT"
-	IndexFaissHNSWPQ  IndexType = "FAISS_HNSW_PQ"
-	IndexFaissHNSWSQ  IndexType = "FAISS_HNSW_SQ"
-	IndexFaissHNSWPRQ IndexType = "FAISS_HNSW_PRQ"
+	IndexVector IndexType = "VECINDEX"
```
Comment (czs007) [inline internal/util/indexparamcheck/index_type.go:27]: TODO: remove this definition

**Pull request merged**